### PR TITLE
Watermark for HeroImage on Social Sharing

### PR DESCRIPTION
**Pull request by VeenaYemmiganur**
# Description
Generate social share watermark for stories

Feature:
Bold now supports adding an image watermarking

Added watermark on story hero image - on social sharing

Fixes # (issue-reference)
https://github.com/quintype/ace-planning/issues/754

- [ ] New feature (non-breaking change which adds functionality)

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/image-tags.js [hunk 1]
```diff
@@ -112,6 +112,10 @@ function pickImage({ pageType, config, seoConfig, data, url }) {
  */
 export function ImageTags(seoConfig, config, pageType, data, { url = {} }) {
   const { image, alt, includesHost = false } = pickImage({ pageType, data, url, seoConfig, config });
+  const story = get(data, ["data", "story"]);
+  const fallbackValue = story ? false : true;
```
Comment (Jeevan-Kishore) [inline src/image-tags.js:116]: is this a check to see if the object empty?

File: src/image-tags.js [hunk 2]
```diff
@@ -123,32 +127,47 @@ export function ImageTags(seoConfig, config, pageType, data, { url = {} }) {
...
       name: "twitter:image",
-      content: includesHost
-        ? image
-        : `https://${config["cdn-image"]}/${image.path([16, 9], {
-            w: 1200,
-            auto: "format,compress",
-            ogImage: true,
-            enlarge: true,
-          })}`,
+      content: isWatermarkDisabled ? actualImage([16, 9]) : watermarkImage([40, 21]),
```
Comment (Jeevan-Kishore) [inline src/image-tags.js:162]: DRY, can this be a function call? since its used across places?

**Review by Jeevan-Kishore — CHANGES_REQUESTED** [review]

File: src/image-tags.js (hunk 1 above)
Comment (VeenaYemmiganur) [reply src/image-tags.js:116]: Tests were failing so added this  to check if its a "collection" or "story"

File: src/image-tags.js [hunk 3]
```diff
@@ -112,6 +113,12 @@ function pickImage({ pageType, config, seoConfig, data, url }) {
  */
 export function ImageTags(seoConfig, config, pageType, data, { url = {} }) {
   const { image, alt, includesHost = false } = pickImage({ pageType, data, url, seoConfig, config });
+  const story = get(data, ["data", "story"]);
+  const publisherConfig = get(data, ["config", "publisher-attributes"], {});
+  const fallbackValue = story ? false : true;
```
Comment (cricket-lover) [inline src/image-tags.js:118]: Can we think of a better logic to handle this? This line of code seems wrongs to me.

**Review by cricket-lover — APPROVED** [review]

File: src/image-tags.js (hunk 3 above)
Comment (VeenaYemmiganur) [reply src/image-tags.js]: ok

**Review by cricket-lover — APPROVED** [review]

**Pull request merged**